tests/Query/Filter/TagsField.test.ts

```diff
@@ -263,4 +263,66 @@ describe('tag/tags', () => {
         expect(filter).not.toMatchTaskFromLine('- [ ] stuff #HOME');
         expect(filter).not.toMatchTaskFromLine('- [ ] stuff #work #HOME'); // searches multiple tags
     });
+
+    describe('filtering with all positive variants', () => {
+        test.concurrent.each([
+            'tag matches',
+            'tag regex matches',
+            'tag regexp matches',
+            'tag matches regex',
+            'tag matches regexp',
+            'tags matches',
+            'tags regex matches',
+            'tags regexp matches',
+            'tags matches regex',
+            'tags matches regexp',
+            'tags match',
+            'tags regex match',
+            'tags regexp match',
+            'tags match regex',
+            'tags match regexp',
+        ])('%s short tag', (cmd) => {
+            // Arrange
+            const filter = new TagsField().createFilterOrErrorMessage(String.raw`${cmd} /#foo/`);
+
+            // Act, Assert
+            expect(filter).toMatchTaskFromLine('- [ ] Do #foo stuff');
+            expect(filter).not.toMatchTaskFromLine('- [ ] Do foo stuff');
+            expect(filter).not.toMatchTaskFromLine('- [ ] Do #bar stuff');
+            expect(filter).toMatchTaskFromLine('- [ ] Do #bar and some #foo stuff');
+            expect(filter).toMatchTaskFromLine('- [ ] Do #foobar stuff');
+            expect(filter).not.toMatchTaskFromLine('- [ ] Do #barfoo stuff');
+        });
+    });
+
+    describe('filtering with all negative variants', () => {
+        test.concurrent.each([
+            'tag does not match',
+            'tag regex does not match',
+            'tag regexp does not match',
+            'tag does not match regex',
+            'tag does not match match regexp',
+            'tags does not match',
+            'tags regex does not match',
+            'tags regexp does not match',
+            'tags does not match regex',
+            'tags does not match regexp',
+            'tags do not match',
+            'tags regex do not match',
+            'tags regexp do not match',
+            'tags do not match regex',
+            'tags do not match regexp',
```

Comment:
Yes. Even though the "matches regex /.../" variant only works by accident (before and after the change) since the RegexMatcher.validateAndConstruct() method removes any surplus clauses in front of the actual regex. So even "matches garbage /.../" would work - the garbage is silently ignored. Not sure if we should change that.

Reply:
No, these options are not meant to be documented. I would just document the "standard syntax" we agree upon, and then we can mention somewhere that you can also write "regex matches" instead of "matches" and "regexp" instead of "regex". Not listing all the combinations. This is a bit similar to SQL where you can just write "COMMIT" instead of "COMMIT TRANSACTION", both do the same.

Reply:
There is something fundamental that I am not understanding...

I find that the following give errors currently:

```
description matches /Fred/
description matches garbage /Fred/
```

Whereas this works:

```
description regex matches /Fred/
```

Reply:
> No, these options are not meant to be documented. I would just document the "standard syntax" we agree upon, and then we can mention somewhere that you can also write "regex matches" instead of "matches" and "regexp" instead of "regex". Not listing all the combinations. This is a bit similar to SQL where you can just write "COMMIT" instead of "COMMIT TRANSACTION", both do the same.

The purpose of the tests is to prevent developers breaking things in future.

If there was a previously allowed syntax that is not currently covered by tests, then it probably should be added to the tests, in case a user had used it, without realising it was unintended behaviour.

But for the regex case, I do not believe there are any such undocumented cases.

If there are, please can we talk about them in #1143 - where users who gave feedback on that one already will be aware of the discussion, and they will not get lost amongst the discussion here about implementation details.

Reply:
We should also discuss in #1143 what the preferred new "standard syntax" will be.

Reply:
> I find that the following give errors currently:

With "currently" do you mean the current PR branch or the current main branch? In the PR branch I do not see errors, and this variant is also explicitly tested to work. In the main branch it does not work because the regex clause is currently mandatory. Everything that comes after the "matches" and before the first slash of the regex is silently ignored (in the PR and the main branch).

Reply:
> But for the regex case, I do not believe there are any such undocumented cases.

The fact that `regex matches garbage /Fred/` works was probably not intended and is currently neither documented nor tested in the main branch and the PR. Not sure if we should consider it a bug or a feature. That's why I did not change or test this behavior, except for testing but not documenting the special case where `garbage` is `regex`, i.e. allowing to postpone the regex clause. Yes, we should discuss this in #1143.